ui/lib/kv/addon/components/kv-patch-editor.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { A } from '@ember/array';
+import { hasWhitespace, isNonString } from 'vault/utils/validators';
+
+const WHITESPACE_WARNING =
```

Comment:
Would these make sense to live next to the validators themselves? I know we have the whitespace warning at least one other place

Reply:
Oh yeah! I can do that 😄